data/console/test_openssl_nodejs.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+set -e
+
+test_result_wrapper(){
+  local VERSION=$1
+  local FILE=$2
+
+  local NODE="/usr/bin/node$VERSION"
+  local OUTPUT="/tmp/test_output"
+
+  # Run the test (using common and custom flags if present) and save the output
+  echo "Running $NODE $GLOBAL_FLAGS ${node_flags[$VERSION $FILE]}$FILE"
+  set +e
+  $NODE $GLOBAL_FLAGS ${node_flags[$VERSION $FILE]} $FILE &> $OUTPUT
+  RESULT=$?
+  set -e
+
+  # If test failed, keep track of failure and print out its output
+  if [ $RESULT -ne 0 ]; then
+    echo "FAILED"
+    echo "Test Output:"
+    cat $OUTPUT
+
+    if [ "${skip_test[$VERSION $FILE $OS_VERSION]}X" = "skipX" ]; then
+      echo "Test was in the exclusion list. Failed result will be SKIPPED."
+      echo "Node v$VERSION Test $FILE" >> $SKIPPED_TEST_LIST
+    else
+      echo "Node v$VERSION Test $FILE" >> $FAILED_TEST_LIST
+    fi
+  else
+    echo "OK"
+  fi
+  rm $OUTPUT
+}
+
+test_node_version(){
+  local VERSION="$1"
+
+  echo "Starting to test node version $VERSION"
+
+  local SOURCE_FILE=""
+  SOURCE_FILE=$(cd /usr/src/packages/SOURCES/ && find . -type f -iname "node-v$VERSION*.tar.xz")
+  SOURCE_FILE=${SOURCE_FILE:2} # strip ./ from name
+
+  if [ -z "$SOURCE_FILE" ]; then
+    echo "Can't find the sources of nodejs$VERSION"
+    exit 1
+  fi
+
+  local SOURCE_DIR
+  SOURCE_DIR=${SOURCE_FILE%.tar.xz} #remove extension from filename to get directory
+
+  # Unpack and apply patches to source
+  # Run the tests on the source using the installed binary
+  pushd "/usr/src/packages/SOURCES"
+    quilt setup "../SPECS/nodejs$VERSION.spec"
+    pushd "$SOURCE_DIR"
+      quilt push -a
+      # Run all test-crypto and test-tls available
+      for f in $(find test \( -path \*sequential\* -or -path \*parallel\* \) \( -name test-crypto-\* -or -name test-tls-\* \)); do
+        test_result_wrapper $VERSION $f
+      done
+    popd
+  popd
+
+  # Cleanup sources
+  rm -rf /usr/src/packages/SOURCES/*
+  rm -rf /usr/src/packages/SPECS/* 
+}
+
+main(){
+
+  OS_VERSION="$1"
+  echo "OS_VERSION: $OS_VERSION"
+
+  # Install dependencies to apply source patches and run tests
+  zypper -n in quilt rpm-build openssl-1_1
```

Comment:
I think it's safe to make sure that `openssl` is installed and only install `openssl-1_1` in SLE12 SP4+ systems. This can be in addition to `openssl`

Reply:
On 12-SP4/5 we need openssl 1.1 (or newer I guess), instead of the default 1.0 that comes from `openssl` package.

What about installing 1.1 only on 12-SP4/5? 
Or maybe only after checking that 1.0 is installed.

Reply:
yes, having version specific code is fine. I would prefer if the default is version-less